Src/FluentAssertions/Xml/Equivalency/XmlReaderValidator.cs

```diff
@@ -209,5 +226,27 @@ private Failure ValidateText()
 
             return null;
         }
+
+        private Failure ValidateAgainstNulls()
+        {
+            if ((expectationReader is null) ^ (subjectReader is null))
```

Comment:
🤔 I think it would be better to use more ordinary `&&`, `||` and negation instead of using xor. I had to look up this operator to remind myself what it did again.

Reply:
🤡 I Just copied it from https://github.com/fluentassertions/fluentassertions/blob/develop/Src/FluentAssertions/Primitives/StringValidator.cs

Reply:
It's only 20 months since you looked it up last time 🙃 
https://github.com/fluentassertions/fluentassertions/pull/1141#discussion_r326902258

Would changing it to 
```cs
if ((expectationReader is null) != (subjectReader is null))
```
be more readable to you?
It's an equivalent way to expresses "one, but not both".

Reply:
20 months?? I don't even remember what I did last week ;-). 

But yes, much more expressive.

Reply:
FYI, I've changed the other occurences of `^` to `!=` in #1574.

Reply:
Changed